app/models/volunteer.rb

```diff
@@ -102,11 +106,18 @@ class Volunteer < ApplicationRecord
   scope :loj_active_take_more, lambda {
     accepted_joined.will_take_more_assignments.where(assignments: { id: Assignment.active.ids })
   }
+
   scope :seeking_clients, lambda {
     accepted_joined
       .merge(Assignment.inactive)
       .where.not(assignments: { volunteer_id: with_active_assignments.ids })
       .or(loj_without_assignments)
+  }
+  scope :seeking_clients_will_take_more, lambda {
+    accepted_joined
+      .merge(Assignment.inactive)
+      .where.not(assignments: { volunteer_id: with_active_assignments.ids })
+      .or(loj_without_assignments)
       .or(loj_active_take_more)
```

Comment:
Not sure if the combination of not/or/or will really do what you want, maybe it does though ;-)

Reply:
I made extensive tests that made me believe it works. But I'm obviously open to better solutions. Incompatible queries made this scopes to be quite a journey. 😉